### DISCUSSION TITLE
DUSD - RTTY

### DISCUSSION THREAD

**Discussion by RobbiePh** [foldynl/QLog#299] [General]
Hi there,

Just installed and using Qlog.
Is there anyway I can get DUSB to display instead of RTTY?

By the way I am using SunDSR2 DX, and in the rig I have selected Kenwood TS-480.
It is a limitation of the rig I selected?

Any support for the SunSDR radios? I flex, ana, thesis etc?

![image](https://github.com/foldynl/QLog/assets/96705124/b09077a8-1ab3-4e1f-a23c-3b8d4738e2e7)

De Ei2iP

**Comment by RobbiePh** [comment]
![image](https://github.com/foldynl/QLog/assets/96705124/2f98d422-ddbd-4047-9d2b-d5b4974aaf9a)

**Comment by foldynl** [comment]
From README file:

"QLog supports all rigs supported by Hamlib" 

As far as I know, Hamlib does not natively support SunSDR. There are a few hacks on how to do this, but no native support.

If you chose an alternative rig, it is only a workaround, which carries with it various pitfalls. In your case, you can't see your favorite mode. Mods in QLog are derived from the selected rig. In your case of the TS-480, hamlib only defines the mods what you can see in QLog

from hamlib source:

`#define TS480_ALL_MODES (RIG_MODE_AM|RIG_MODE_CW|RIG_MODE_CWR|RIG_MODE_SSB|RIG_MODE_FM|RIG_MODE_RTTY|RIG_MODE_RTTYR)`

I can see following solutions:

1) to help QLog community to implement TCI
2) use https://github.com/ftl/tciadapter

**Reply by rsaue** [reply]
Hi, I am using TCI adapter when connecting Qlog to a SunSDR2-DX with EESDR v 3 on linux and Windows. Works very well.

73 de LB2EG Richard

**Reply by foldynl** [reply]
thanks for your info. Good to know there is a workaround.

**Comment by RobbiePh** [comment]
Thank you for your reply

Would you mind sharing a screen shot of your rig settings menu?
Which rig model did you select? Port and ip address.

I installed the TCI Adaptor.

De Ei2iP

**Reply by RobbiePh** [reply]
I thought as much, so you do not have TCI connecting to your SUNSDR and displaying the Data modes. It is not working for you?

It is easy connect the SUNsdr to Qlog, they problem is not showing the datamodes, which means I have to edit the contact in the logbook all the time...

**Comment by RobbiePh** [comment]
Here is mine.. I don’t see the digital modes DUSB?

SUNsdr is connected.

![image](https://github.com/foldynl/QLog/assets/96705124/ec66f435-d883-47fa-bfed-1420c0b5eb17)

![image](https://github.com/foldynl/QLog/assets/96705124/a25fdf05-876e-4272-b10a-a616e5773bc7)


De Ei2iP

**Reply by rsaue** [reply]
I have now reviewed the original post and subsequent postings, as i did not see the screenshots reading the message on my phone earlier on. 

You are using ESDR Radiosync which emulates a TS-480. Your setup is okay, and so are the modes shown in Qlog. 

I see the same on my site using Radiosync. 

Hamlib, which QLog uses for CAT translates EESDR mode DIGL to RTTY, and DIGU to RTTYR. Changing mode to RTTYR in QLog changes ESSDR to DIGU. All this is OK, as TS-480 which Radiosync emulates does not know any other digital mode than FSK (= RTTY). 

If you as an alternative uses FLDIGI as an alternative to Radiosync, FLDIGI uses TCI to and from ESDR software, and you  connect to FLDIGI using QLog Hamlib CAT. FLDIG translates DIGU as PKTUSB and DIGL as PKTLSB. As far as I can remember this naming is on par with the Yaesu standard for digimodes some years ago. 

Using TCIadapter, as I suggested in my original posting is not a better option. TCIadapter taps into Hamlib NEt rigctl . Net Rigctl only knows AM/FM/CW/USB/LSB.  No digital modes. 

I would say that you have the easiest and most convenient CAT setup with the setup you have now. 

You will seldom log a Digimode QSO manually in QLog. And newer as a DIGU or DIGL QSO rather than FT8, RTTY, PSK31, etc. 

When working with digital modes, you will usually log the contacts in the digital mode program log program, and import those to QLog with UDP connection (or ADIF import) and this gives you the correct mode in log too... 

73 de LB2EG Richard

**Comment by RobbiePh** [comment]
I have tried everything and no luck with the DUSB.. For the moment, I will park Glog.

Thanks for all the help.


De Ei2iP

**Reply by rsaue** [reply]
I honestly do not se this as an adequate show-stopper, as NOBODY usually log a digital mode QSO as either DIGU (or DIGL). There is no digital mode called DIGU (or DIGL). DIGU is only RIG speak for USB without the stuff you usually employ when using the microphone (Gain, Compression, equalizer etc..). 

Anyway, good luck on the bands and Good DX!

73 de LB2EG, Richard

**Reply by RobbiePh** [reply]
Yes I agree, I see now that I can change the mode on the logging screen, so I agree this is not a show-stopper.
All is ok now on the DUSB, change to FT8 on screen is easy.

![image](https://github.com/foldynl/QLog/assets/96705124/2eca288d-6ba5-46c8-8e07-e0a0678d3c1b)


Now I need to see if Qlog can receive UDP messages from either JTDX or Grid Tracker, I will check this out.

Thanks for your help so far, de Ei2iP.

**Reply by rsaue** [reply]
Good! 

Just check the Qlog wiki... 

73 de LB2EG Richard 
![Skjermbilde fra 2024-01-15 19-55-19](https://github.com/foldynl/QLog/assets/22942440/0fc017e0-5129-4a2a-b8dd-9b4f35297d65)

**Comment by foldynl** [comment]
I have a question. Is there anything that simulates Sunsdr rig? I went through the TCI protocol specification and it's nothing complicated. Unfortunately, without this rig it will be difficult to debug the QLog integration. 

That's why I think about how it could be simulated without a rig. Or are there any "public" or "paid" access to these rigs? I am only concerned with debugging, which usually takes 2-3 days.

**Reply by rsaue** [reply]
Hi Ladislav,
No SunSDR simulator as far as I know. I have a SunSDR2-DX with a working setup on windows and Linux. I can do some testing on Linux (where I can compile test code) if You want. 

The Thetis software for ANAN-radios, Hermes Lite, and some other radios use a subset of TCI protcol ( no audio and no CW) good enough for ordinary Rig Control. (I have an ANAN-200D and a Hermes Lite2).  If you know a local ham with one of those radios, you can use Thetis as a testing environment. 

73 de LB2EG Richard

**Reply by foldynl** [reply]
Richard, could I ask you to test the initial attempt to implement TCI in QLog? I tried Thetis, but I don't have the required HW and I don't know how to configure it correctly, because Thetis closes my connection at the start. 

Can you please try the `TCISupport` branch, which is built on top of the new QLog 0.32 version?

please, enabled debugging for TCI Driver before running QLog

`export QT_LOGGING_RULES=qlog.rig.driver.tcidrv.*=true` and then run `qlog > debug.log`

or one-liner
`QT_LOGGING_RULES=qlog.rig.driver.tcidrv.*=true ./qlog > debug.log`

and provide me the log file?

**Reply by rsaue** [reply]
Hi, 
compiled and tested on ubuntu linux 23.10 and qt5. Its working with both RX1 and RX2, but the debug.log is empty... Will test again in the evening and report back again... 

One thing, in the logging section, CW is no longer an option when using TCI... Lots of other modes but no CW... 

73 de LB2EG Richard

**Comment by foldynl** [comment]
I'm sorry my mistake. the display should have been this

`QT_LOGGING_RULES=qlog.rig.driver.tcidrv.*=true ./qlog 2>debug.log`

Missing CW is something what I have to investigate based on `debug.log`. QLog should show all modes what are received from RIG during the initial part of connection. If CW is missing then it has to be investigated.

**Reply by rsaue** [reply]
Hi, 
tested again and now the debug log is filled up. Qlog responds nicely to mode and QRG changes on the radio (EESDR3 v. 1.0.7 beta), and the radio responds to mode and QRG changes initiated from QLOG. The missing CW in log window is up to you to investigate. The QRG display in Qlog shows CW and acts on changing mode to CW. The mode selection in Qlog Logging section responds to other mode changes in radio except from CW which is not shown there. Changes in mode selection done in Qlog Logging section changes also mode on the radio. 

One observation: the RF power indicator in the Qlog CAT window does not respond to Power changes on the radio. It does, however, show the level of the radio Tune function. 

debug log is attached. 

Attached is also the logfile on TCI commands on radio TCI server for the session. 


73 de LB2EG Richard

[debug.log](https://github.com/foldynl/QLog/files/14283914/debug.log)

[ESDR TCI server log.txt](https://github.com/foldynl/QLog/files/14284066/ESDR.TCI.server.log.txt)

**Comment by foldynl** [comment]
Many thanks, this is exactly what I needed  together with the TCI server log, there are many things to research.  it seems that the direction is correct and it is only necessary to fine-tune some things.

As for the missing CW, that's a mystery.  Sorry for a simple question, but do you have active CW set in QLog?? Am I understanding correctly that you don't see the CW where the following picture shows?

![image](https://github.com/foldynl/QLog/assets/942769/eeef429c-a7d3-4efb-bb6c-e69420c8056f)

**Reply by rsaue** [reply]
Hi,
mystery solved. Mea Culpa. Looking through the setup menus, I found that CW was marked as "Disabled" in settings->Mode menu. Don't know how this happened, but probably due to finger trouble at my end at some point ...   

73 de LB2EG Richard
